shell/artifacts/Profile/source/FavoriteFoodPicker.js

```diff
@@ -6,80 +6,115 @@
+    Eggs: '1f373',
+  };
+
+  const template = html`
 
-  let styles = `
 <style>
   [${host}] {
-    padding: 6px 0;
+    padding: 24px;
     text-align: center;
   }
-  [${host}] > * {
-    vertical-align: middle;
-  }
-  [${host}] select {
-    padding: 6px;
-    font-size: 14px;
+  [${host}] > [foods] {
+    display: inline-block;
   }
-  [${host}] .x-select {
+  [${host}] > [foods] > div {
     display: inline-block;
-    position: relative;
+    width: 88px;
+    height: 88px;
+    padding: 16px;
+    color: silver;
+    border-radius: 16px;
+    text-align: center;
   }
-  [${host}] .x-select::after {
-    content: '▼';
+  [${host}] > [foods] > div[selected] {
+    background: #eeeeee;
+    color: gray;
+  }
+  [${host}] > [foods] > div > * {
     display: block;
-    position: absolute;
-    right: 8px;
-    bottom: 6px;
-    transform: scaleY(0.6);
-    pointer-events: none;
   }
-  [${host}] .x-select > select {
-    position: relative;
-    margin: 0;
-    padding: 8px 24px 8px 6px;
-    border: 0;
-    border-bottom: 1px solid #ddd;
-    background-color: transparent;
-    border-radius: 0;
-    font-size: 14px;
-    font-weight: 300;
-    overflow: hidden;
-    outline: none;
-    -webkit-appearance: none;
+  [${host}] > [foods] > div img {
+    display: block;
+    width: 64px;
+    margin: 0 auto 16px;
   }
 </style>
-  `;
 
-  let template = `
-${styles}
 <div ${host}>
-  <div class="x-select">
-    My favorite food is:
-    <select on-change="_onFavoriteFoodChanged">
-      ${['Pizza', 'Bacon', 'Lobster', 'Tim Tam', 'Kale']
-        .map(i => `<option value="${i}" selected$={{selected${i}}}>${i}</option>`).join('')}
-    </select>
+  <div selector>
+    My favorite Food
   </div>
+  <div foods>{{foods}}</div>
+  <template foodTemplate>
+    <div value="{{name}}" selected$="{{selected}}" on-click="onSelectFood">
+      <img src="{{src}}">
+      <div>{{name}}</div>
+    </div>
+  </template>
 </div>
+
     `.trim();
 
   return class extends DomParticle {
     get template() {
       return template;
     }
     render(props, state) {
-      let food = props.food && props.food && props.food.food || '';
+      const favorite = props.food && props.food.food;
+      const path = resolver('FavoriteFoodPicker');
+      const models = Object.keys(foods).map(name => {
+        return {
+          name: name,
+          src: `${path}/../assets/noto-emoji-128/emoji_u${foods[name]}.png`,
+          selected: name === favorite
+        };
+      });
       return {
-        [`selected${food}`]: true
+        foods: {$template: 'foodTemplate', models}
       };
     }
+    setFavoriteFood(food) {
+      const foodHandle = this.handles.get('food');
+      foodHandle.set(new foodHandle.entityClass({food}));
+    }
     _onFavoriteFoodChanged(e, state) {
-      const food = this._views.get('food');
-
-      food.set(new food.entityClass({food: e.data.value}));
+      this.setFavoriteFood(e.data.value);
+    }
+    onSelectFood(e) {
```

Comment:
We've been prefixing callbacks with `_` - is this a change or a different use?

Reply:
Particles don't need to differentiate between public and protected API, so we are only using underscores for private fields. This is distinct from the AppShell, where, technically anyway, components have public, protected, and private APIs, and we use single-underscore to indicate protected (double-underscore for private).

Reply:
Cool thanks! :+1: